src/LinuxTracing/LinuxTracingIntegrationTestPuppet.cpp

```diff
@@ -27,7 +27,7 @@ using PuppetConstants = LinuxTracingIntegrationTestPuppetConstants;
 
 static void SleepRepeatedly() {
   for (uint64_t i = 0; i < PuppetConstants::kSleepCount; ++i) {
-    absl::SleepFor(absl::Microseconds(10));
+    absl::SleepFor(absl::Microseconds(100));
```

Comment:
There are multiple layers here:
1. absl
2. glibc
3. kernel

I think we might want to call nanosleep here for the starters and see if it returns EINTR, and it case it does repeat. It could be that abseil here is doing smart think by checking if time is already passed when EINTR is returned and return with noop. and 10 microseconds could be just at the limit here.

worth a try?

Reply:
(and probably add a comment about why the test uses nanosleep instead of absl)

Reply:
> I think we might want to call nanosleep here for the starters and see if it returns EINTR, and it case it does repeat.

This made sense to me but doesn't work. Even when the test fails I don't see `EINTR` ever being returned. After all, why would we expect the target to receive signals?

By the way, what Abseil does is here: https://chromium.googlesource.com/external/github.com/abseil/abseil-cpp/+/HEAD/absl/time/clock.cc#564

Reply:
ok, nevermind. I was under mistaken impression that nanosleep syscall operates in duration - but it actually uses timespec. which means that increasing timeout seems to be the only way here.